File: leet/longest_common_prefix 14/longest_common_prefix.dart
class Solution {
  String longestCommonPrefix(List<String> strs) {
    String long = '';
    int f = 0;
    for (int i = 0; i < strs[0].length; i++) {
      String sub = strs[0].substring(0, i + 1);
      for (int j = 1; j < strs.length; j++) {
        if (strs[j].isEmpty || strs[j].length < sub.length) {
          f = 1;
          break;
        }
        String chq = strs[j].substring(0, i + 1);
        if (sub != chq) {
          f = 1;
          break;
        }
      }
      if (f == 1) break;
      long = long.length < sub.length ? sub : long;
    }
    return long;
  }
}

File: leet/longest_common_prefix 14/longest_common_prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    var longString = '';
    var flag = 0;
    for (var i = 0; i < strs[0].length; i++) {
        var subString = strs[0].substring(0, i + 1);
        for (var j = 1; j < strs.length; j++) {
            if ((strs[j] == '') || (strs[j].length < subString.length)) {
                flag = 1;
                break;
            }
            if (subString !== strs[j].substring(0, i + 1)) {
                flag = 1;
                break;
            }
        }
        if (flag === 1) break;
        longString = longString.length < subString.length ? subString : longString;
    }
    return longString;
};

console.log('ans -- > ' + longestCommonPrefix(["flower","flow","flight"]));